diff --git a/includes/entity.inline_entity_form.inc b/includes/entity.inline_entity_form.inc
index 322cd74..3e0c1e0 100644
--- a/includes/entity.inline_entity_form.inc
+++ b/includes/entity.inline_entity_form.inc
@@ -7,7 +7,7 @@
 
 class EntityInlineEntityFormController {
 
-  protected $entityType;
+  public $entityType;
   public $settings;
 
   public function __construct($entityType, array $settings) {
diff --git a/inline_entity_form.module b/inline_entity_form.module
index 2ca0501..aee3e19 100644
--- a/inline_entity_form.module
+++ b/inline_entity_form.module
@@ -230,6 +230,7 @@ function inline_entity_form_field_widget_settings_form($field, $instance) {
   // Add entity type specific settings if they exist.
   $settings_form = $controller->settingsForm($field, $instance);
   if (!empty($settings_form)) {
+    $entity_info = entity_get_info($controller->entityType);
     $element['type_settings'] = array(
       '#type' => 'fieldset',
       '#title' => t('Inline Entity Form: %type', array('%type' => $entity_info['label'])),
